documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -690,6 +690,27 @@ private CarMapper mapper;
 
 A mapper which uses other mapper classes (see <<invoking-other-mappers>>) will obtain these mappers using the configured component model. So if `CarMapper` from the previous example was using another mapper, this other mapper would have to be an injectable CDI bean as well.
 
+[[injection-strategy]]
+=== Injection strategy
+
+When using <<using-dependency-injection,dependency injection>>, you can choose between field and constructor injection. This can be done by either providing the injection strategy via __Mapper__ or __MapperConfig__ annotation.
+
+.Obtaining a mapper via dependency injection
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper(componentModel = "cdi", uses = EngineMapper.class, injectionStrategy = InjectionStrategy.CONSTRUCTOR)
+public interface CarMapper {
+    CarDto carToCarDto(Car car);
+}
+----
+====
+
+The generated mapper will inject all classes defined in the **uses** attribute. When **InjectionStrategy.CONSTRUCTOR** is used, the constructor will have the appropriate annotation and the fields won't. When **InjectionStrategy.FIELD** is used, the annotation is on the field itself. For now, the default injection strategy is field injection. It is recommended to use constructor injection to simplify testing.
+
+If you inject additional fields by hand, we recommend using setter injection.
```

Comment:
I would make this a TIP. And I think that we should say that for abstract classes or decorators that setter injection should be used.